### PR TITLE
Feature/gds assessment select main service page

### PR DESCRIPTION
**Pull request by omarCreativeDev**
- make `select-main-service` page gds compliant

### CODE REVIEW

**Comment by omarCreativeDev** [comment]
[select main service page](https://trello.com/c/miuqTTDY/458-select-main-service-page)

File: src/app/features/registration/select-main-service/select-main-service.component.ts [hunk 1]
```diff
@@ -1,144 +1,129 @@
...
+        (error: HttpErrorResponse) => {
+          this.serverError = this.errorSummaryService.getServerErrorMessage(error.status, this.serverErrorsMap);
+          this.errorSummaryService.scrollToErrorSummary();
+        }
+      )
+    );
+  }
 
-    this._registrationService.updateState(this.registration);
+  public onSubmit(): void {
```
Comment (ChazUK) [inline src/app/features/registration/select-main-service/select-main-service.component.ts]: Should the data not be being saved to the state in the section?

File: src/app/core/model/location.model.ts [hunk 2]
```diff
@@ -1,15 +1,23 @@
...
+  county: string;
+  isRegulated: boolean;
   locationId: string;
-  postcodeNo: string;
+  locationName: string;
+  mainService: string;
+  postalCode: string;
+  townCity: string;
+  // TODO check if this user object can be moved
+  user: {
```
[the remark this answers is not in the document]
Comment (omarCreativeDev) [reply src/app/core/model/location.model.ts]: yea i agree, it was originally there as part of the registration interface but when i take it out lots of other components and pages that I haven't worked on start throwing up errors so i re added it with a TODO to clean up when i tackle the next pages. im conscious of this pr getting too big

File: src/app/core/model/service.model.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+export interface Service {
```
[the remark this answers is not in the document]
Comment (omarCreativeDev) [reply src/app/core/model/service.model.ts]: done

File: src/app/core/model/category.model.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import { Service } from '@core/model/service.model';
+
+export interface Category {
```
[the remark this answers is not in the document]
Comment (omarCreativeDev) [reply src/app/core/model/category.model.ts]: done

File: src/app/features/registration/select-main-service/select-main-service.component.ts (hunk 1 above)
Comment (omarCreativeDev) [reply src/app/features/registration/select-main-service/select-main-service.component.ts]: done

**Pull request merged**